### ISSUE TITLE
Explore libraries for push/email notifications

### ISSUE THREAD

**Issue by milanmitreski** [sntntn/FitPlusPlus#22]
Sending push and email notifications will be implemented with some libraries. Exploring what libraries are available and comparing them so that the libraries that are most adequate for our project are chosen.

**Comment by milanmitreski** [comment]
Email notifications are more simpler: library `FluentEmail` is suggested. 

As for the push notifications, the situation is a little bit more complicated. The solution is the following (this should be considered for EventBus message modelling also): each service should send a list of users (clients, trainers, managers) along with the notification it sends. The notifications will be stored in a notification section (on the frontend). It is needed to track whether the notification has been read by the user or no (in this solution, we will only send out an email and a push notification that there is a new notification waiting). There is a way to show push notifications even if the user is not logged in.

All-in-all: All notification will be stored inside the notification section of the frontend, with the possibility of push and email notifications sent that a new notification has arrived for the user (and that notification can be seen inside of his notification section in his account).

As this is an important modelling decision, I think all of us should agree on this solution.

@ivvvvvva @stemil01 @labovicca @sntntn @NataMata pending approval.

**Comment by sntntn** [comment]
I fully agree with the suggested technologies (FluentEmail for email and FirebaseAdmin for push notifications). After reviewing the notification strategy, I believe the following approach for my case (ChatService) would provide the best user experience::

Email notifications: These should be sent only for key events, such as unlocking a 30-day session(paid mentorship). This ensures the emails are meaningful and avoid spamming the user.
Push notifications: Each chat message should trigger a push notification informing the user about a new message in their chat session. For example:
Message: "You have a new message in your chat session with [Other Participant's Name]."
Frontend Notification Section: All notifications, including chat-related ones, should still be logged in the notification section on the frontend for users to review later.
This approach balances user experience and system efficiency while aligning perfectly with the suggested technologies.

Let me know if you see any challenges or adjustments needed for integrating this into the Notification Service.
